index.d.ts

```diff
@@ -41,15 +43,18 @@ export interface Action {
  *
  * *Do not put API calls into reducers.*
  *
- * @template S State object type.
+ * @template S The type of state consumed and produced by this reducer.
+ * @template A The type of actions the reducer can potentially respond to.
  */
-export type Reducer<S> = <A extends Action>(state: S | undefined, action: A) => S;
+export type Reducer<S = {}, A extends Action = Action> = (state: S | undefined, action: A) => S;
 
 /**
  * Object whose values correspond to different reducer functions.
+ *
+ * @template A The type of actions the reducers can potentially respond to.
  */
-export type ReducersMapObject<S> = {
-  [K in keyof S]: Reducer<S[K]>;
+export type ReducersMapObject<S = {}, A extends Action = Action> = {
```

Comment:
Default for `S` here is unclear for me too.

Reply:
Having defaults means you can just reference `ReducersMapObject` if you don't care about the types involved. That tends to happen if you have a generic type parameter `T extends ReducersMapObject` for example.

Reply:
This makes some sense. But wouldn't `ReducersMapObject<{}>` be the same as just `{}`? Therefore `T extends {}` only means that `T` is not `null` or `undefined`.

Reply:
You raise a good point and I think it means I chose the wrong default... instead it should be `{ [k: string]: any }`. That way `ReducersMapObject` should be equivalent to `{ [k: string]: Reducer<any> }`?

Reply:
Yeah, I think that should work.